lbrynet/wallet/resolve.py

```diff
@@ -284,87 +253,56 @@ def prepare_claim_queries(start_position, query_size, channel_claim_infos):
         start_position = (page - 1) * page_size
         queries, names, claim_positions = self.prepare_claim_queries(start_position, page_size,
                                                                      channel_claim_infos)
-        page_generator = await self.iter_channel_claims_pages(queries, claim_positions, names,
-                                                              certificate, page_size=page_size)
         upper_bound = len(claim_positions)
         if not page:
             return None, upper_bound
         if start_position > upper_bound:
             raise IndexError("claim %i greater than max %i" % (start_position, upper_bound))
+        page_generator = await self.iter_channel_claims_pages(queries, claim_positions, names,
+                                                              certificate, page_size=page_size)
         return page_generator, upper_bound
 
 
-# Format amount to be decimal encoded string
-# Format value to be hex encoded string
-# TODO: refactor. Came from lbryum, there could be another part of torba doing it
-def format_amount_value(obj):
-    if isinstance(obj, dict):
-        for k, v in obj.items():
-            if k in ('amount', 'effective_amount'):
-                if not isinstance(obj[k], float):
-                    obj[k] = dewies_to_lbc(obj[k])
-            elif k == 'supports' and isinstance(v, list):
-                obj[k] = [{'txid': txid, 'nout': nout, 'amount': dewies_to_lbc(amount)}
-                          for (txid, nout, amount) in v]
-            elif isinstance(v, (list, dict)):
-                obj[k] = format_amount_value(v)
-    elif isinstance(obj, list):
-        obj = [format_amount_value(o) for o in obj]
-    return obj
-
-
 def _get_permanent_url(claim_result, certificate_id):
     if certificate_id:
         return f"{claim_result['channel_name']}#{certificate_id}/{claim_result['name']}"
     else:
         return f"{claim_result['name']}#{claim_result['claim_id']}"
 
 
-def _verify_proof(name, claim_trie_root, result, height, depth, transaction_class, hash160_to_address):
+def _verify_proof(name, claim_trie_root, result, height, depth, transaction_class, ledger):
     """
     Verify proof for name claim
     """
+    support_amount = sum([amt for (stxid, snout, amt) in result['supports']])
 
-    def _build_response(name, value, claim_id, txid, n, amount, effective_amount,
-                        claim_sequence, claim_address, supports):
+    def _build_response(name, tx, nOut):
+        output = tx.outputs[nOut]
         r = {
             'name': name,
-            'value': hexlify(value),
-            'claim_id': claim_id,
-            'txid': txid,
-            'nout': n,
-            'amount': amount,
-            'effective_amount': effective_amount,
+            'value': hexlify(output.script.values['claim']),
```

Comment:
Is this `value` actually used by any apps? I imagine we can just drop this from the dictionary. I think the protobuf stuff is isolated in lbrynet and doesn't need to be exposed to UI facing apps.

Reply:
its hex while not decoded, then it becomes the value if we can decode or None and a new field 'hex' is added for displaying the hex value when its unable to decode. I will ping you in Slack to discuss some simplification for that, but yeah its complex and probably unused.